pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -120,16 +137,50 @@ func createRawDeployment(componentMeta metav1.ObjectMeta, workerComponentMeta me
 		addGPUResourceToDeployment(workerDeployment, constants.WorkerContainerName, tensorParallelSize)
 		deploymentList = append(deploymentList, workerDeployment)
 	}
-
-	return deploymentList
+	return deploymentList, nil
 }
 
-func createRawDefaultDeployment(componentMeta metav1.ObjectMeta,
+func createRawDefaultDeployment(clientset kubernetes.Interface, componentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec,
-	podSpec *corev1.PodSpec) *appsv1.Deployment {
+	podSpec *corev1.PodSpec) (*appsv1.Deployment, error) {
 	podMetadata := componentMeta
 	podMetadata.Labels["app"] = constants.GetRawServiceLabel(componentMeta.Name)
 	setDefaultPodSpec(podSpec)
+	var isvcname string
+	var upstreamPort string
+	if val, ok := componentMeta.Labels[constants.InferenceServiceLabel]; ok {
+		isvcname = val
+	} else {
+		isvcname = componentMeta.Name
+	}
+	if val, ok := componentMeta.Labels[constants.ODHKserveRawAuth]; ok && val == "true" {
+		switch {
+		case componentExt != nil && componentExt.Batcher != nil:
```

Comment:
as these two switch cases set the same value for upstreamPort, can we put then inside the same switch case?

Reply:
something like:

```yaml
        - --openshift-service-account=oauth-proxy
        - --client-id=system:serviceaccount:my-namespace:oauth-proxy
        - --client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token
```